R/functions.R

```diff
@@ -395,150 +382,442 @@ vwparams <- function(vwmodel, name) {
+                  weight = NULL, base = NULL, tag = NULL,
+                  multiline = NULL, append = FALSE) {
+    
+    # if namespaces = NULL, define a unique namespace
+    if (is.null(namespaces)) {
+        all_vars <- colnames(data)[!colnames(data) %in% c(targets, probabilities, weight, base, tag)]
+        namespaces <- list(A = list(all_vars))
+    }
+    
+    data <- data.table::copy(data.table::setDT(data))
+    
+    # parse variable names
+    specChar      <- "\\(|\\)|\\||\\:|'"
+    specCharSpace <- "\\(|\\)|\\||\\:| |'"
+    
+    parsingNames <- function(x) {
+        ret <- c()
+        for (el in x)
+            ret <- append(ret, gsub(specCharSpace,'_', el))
+        ret
+    }
+    
+    # parse categorical variables
+    parsingVar <- function(x, keepSpace) {
+        # remove leading and trailing spaces, then remove special characters
+        # then remove isolated underscores.
+        if (!keepSpace)
+            spch <- specCharSpace
+        else
+            spch <- specChar
+        gsub(spch, '_', x)
+    }
+    
+    # Construct labels for multilabel examples: 1:0.3 2:0.3 3:0.3
+    constructLabels <- function(targets, num_row) {
+        # Initialize output vector
+        tmp <- rep("", num_row)
+        # Iterate cost vectors names
+        for(i in 1:length(targets)) {
+            elem_targets <- eval(get(targets[i], envir=parent.frame(n=1)))
+            # Iterate cost vectors individualy (nrow of data)
+            for (j in 1:num_row) {
+                ifelse(is.na(elem_targets[j]),
+                       "",
+                       tmp[j] <- paste(tmp[j],
+                                       paste(i, elem_targets[j], sep = ":")))
+            }
+        }
+        trimws(tmp)
+    }
+    # Construct labels for Context Bandit: 1:0.3:0.3 2:0.3:0.3 3:0.3:0.3
+    constructLabelsCB <- function(targets, probabilities, num_row) {
+        # Initialize output vector
+        tmp <- rep("", num_row)
+        # Iterate cost vectors names
+        for(i in 1:length(targets)) {
+            elem_targets <- eval(get(targets[i], envir=parent.frame(n=1)))
+            elem_probability <- eval(get(probabilities[i], envir=parent.frame(n=1)))
+            # Iterate cost vectors individualy (nrow of data)
+            for (j in 1:num_row) {
+                ifelse(is.na(elem_targets[j]),
+                       "",
+                       tmp[j] <- paste(tmp[j],
+                                       paste(i, elem_targets[j], elem_probability[j], sep = ":")))
+            }
+        }
+        trimws(tmp)
+    }
+    
+    # namespace load with a yaml file
+    if (typeof(namespaces) == "character" && length(namespaces) == 1 &&
+        grepl("yaml$", namespaces)) {
+        # requireNamespace(yaml)
+        print("Using YAML file for loading the namespaces")
+        if (requireNamespace("yaml", quiet=TRUE, as.character=TRUE)) {
+            utils::globalVariable("yaml")
```

Comment:
a) That function does not exist (you missed the plural 's').  Did you run `R CMD check` ?  You should have seen this.

b) Do we really need this?  Is there still an error?  There is no variable 'yaml' here.

Reply:
a) Hmm, strange `R CMD check` didn't show anything. I will take a closer look.

b) maybe actually we don't, I will see if I can rework this quickly

Reply:
I say just remove the line.